core/src/storage/cache.py

```diff
@@ -1,36 +1,68 @@
 from trezor.crypto import random
 
 if False:
-    from typing import Optional
+    from typing import Optional, Dict, List
+
+_MAX_SESSIONS_COUNT = 10
 
 APP_COMMON_SEED = 0
 APP_COMMON_SEED_WITHOUT_PASSPHRASE = 1
 APP_CARDANO_ROOT = 2
 APP_MONERO_LIVE_REFRESH = 3
 
-_cache_session_id = None  # type: Optional[bytes]
-_cache = {}
+_active_session_id = None  # type: Optional[bytes]
+_caches = {}  # type: Dict[bytes, Dict[int, Any]]
+_session_ids = []  # type: List[bytes]
 
 if False:
     from typing import Any
 
 
-def get_session_id() -> bytes:
-    global _cache_session_id
-    if not _cache_session_id:
-        _cache_session_id = random.bytes(32)
-    return _cache_session_id
+def _move_session_ids_queue(session_id: bytes) -> None:
+    # Move the LRU session ids queue.
+    if session_id in _session_ids:
+        _session_ids.remove(session_id)
+    elif len(_session_ids) == _MAX_SESSIONS_COUNT:
```

Comment:
i'd prefer a to call this `while len(_session_ids) >= _MAX_SESSIONS_COUNT:`

Reply:
Mm you mean to protect against a case where we `_MAX_SESSIONS_COUNT + 2` items in `session_ids` for example?

Reply:
`+ 1` is enough for it to be relevant but yes, basically. there seems to be no downside to write the code more defensively, even though this should never happen

Reply:
Yes that was an example :). Ack, coming soon

Reply:
i don't think the `else` is needed